### ISSUE TITLE
Error parsing create table statement with foreign key reference

### ISSUE THREAD

**Issue by cwimberger** [alicebob/sqlittle#30]
I've got a sqlite file with the following table structure:

`CREATE TABLE message_attachment_join ( message_id INTEGER REFERENCES message (ROWID) ON DELETE CASCADE, attachment_id INTEGER REFERENCES attachment (ROWID) ON DELETE CASCADE, UNIQUE(message_id, attachment_id))`

The debug output looks like this:
```
char tok--1 in state-0
lex CREATE(57355)
char tok--1 in state-6
lex TABLE(57387)
char tok--1 in state-13
lex tBare(57392)
char tok--1 in state-11
reduce 6 in:
        state-11
char tok--1 in state-18
lex '('(40)
char tok--1 in state-22
lex tBare(57392)
char tok--1 in state-11
reduce 6 in:
        state-11
char tok--1 in state-26
lex tBare(57392)
char tok--1 in state-11
reduce 6 in:
        state-11
char tok--1 in state-33
lex REFERENCES(57379)
reduce 48 in:
        state-33
char REFERENCES in state-32
reduce 29 in:
        state-32
char REFERENCES in state-42
reduce 44 in:
        state-42
char REFERENCES in state-25
reduce 42 in:
        state-25
char REFERENCES in state-24
reduce 37 in:
        state-24
char REFERENCES in state-31
state-31 saw REFERENCES
error recovery pops state 31
error recovery pops state 24
error recovery pops state 22
error recovery pops state 18
error recovery pops state 13
error recovery pops state 6
error recovery pops state 0
syntax error: unexpected REFERENCES, expecting ',' or ')'

```

**Comment by galteroleg** [comment]
I am facing with same issue, when trying to do Select "db.SelectRowid("name_lookup", 1, "_id")" on table which created by command:
CREATE TABLE name_lookup (data_id INTEGER REFERENCES data(_id) NOT NULL,cars_id INTEGER REFERENCES cars(_id) NOT NULL,normalized_name TEXT NOT NULL,name_type INTEGER NOT NULL,PRIMARY KEY (data_id, normalized_name, name_type)) I get error: syntax error.
I can able to work with database on external tools (I mean database is correct)

**Comment by alicebob** [comment]
> I can able to work with database on external tools (I mean database is correct)  

The way sqlite works is that the original "CREATE TABLE" commands are in the
file, and sqlittle needs to parse them to know what the tables look like. But
the SQL parser in sqlittle doesn't have support for the 'REFERENCES' yet, I
need to add that. Will get to it, but not today :)

**Comment by alicebob** [comment]
Fixed in v1.3.4. I also added support for `CHECK`.

Thanks for reporting and keep them coming!
